tests/test_provider.py

```diff
@@ -14,3 +15,65 @@ def test_report_info_od(provider):
     pinfo2 = provider.get_quota()
     assert pinfo2['used'] > 0
     assert pinfo2['limit'] > 0
+
+def test_interrupted_file_upload(provider):
+    # file_upload_size = 4 MiB, should take 12/4=3 API calls to upload file
+    file_size = 12 * 1024 * 1024
```

Comment:
Is it necessary to have such a large file size? `upload_block_size + 1` might be sufficient. Even if it does need to be 12mb, that should at least be expressed as `upload_block_size * 3` so that this will continue to effectively test interrupted uploads even if we change the upload block size. Consider also that it could work in the test to intentionally reduce the upload block size to 1024, and then the test file could be VERY small, which may save meaningful amounts of time during the testing